cdist/conf/type/__rm/explorer/list

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh -e
+#
+# 2019 Ander Punnar (ander-at-kvlt-dot-ee)
+#
+# This file is part of cdist.
+#
+# cdist is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cdist is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cdist. If not, see <http://www.gnu.org/licenses/>.
+#
+
+if [ -f "$__object/parameter/path" ]
+then
+    path="$( cat "$__object/parameter/path" )"
+else
+    path="/$__object_id"
+fi
+
+[ ! -d "$path" ] && exit 0
+
+remove="$( cat "$__object/parameter/remove" )"
+
+if [ -f "$__object/parameter/exclude" ]
+then
+    exclude="$( cat "$__object/parameter/exclude" )"
+
+    find "$path" -mindepth 1 -maxdepth 1 -path "$remove" \
+        -and -not -path "$exclude"
```

Comment:
@4nd3r mindepth and maxdepth are not by POSIX :)
@telmich Do we allow it anyway?

Reply:
`-and`, `-not` too

Reply:
`-mindepth` is needed to exclude $PWD from list and `-maxdepth` because recursive actions are devil's tools. `-and` and `-not` are just convenience and we can probably get rid of them, but not sure about depth params.

Reply:
and i actually forgot to add note about non-recursivness in manual... :sleepy:

Reply:
even openbsd's `find` support `-mindepth`, `-maxdepth`, `-and` and `-not` so i think we are in the clear here, sans non-recursion note, if you guys agree that we don't want recursivness here...

Reply:
What about `-prune`? ;)

Reply:
Something like
```
$ find .
.
./a
./a/b
./a/b/c
./a/b/c/d
./a/b/c/d/e
./a/b/c/d/e/f
./a/b/c/d/d1
./a/b/c/d/d2
./a/b/c/d/a1
./a/b/c/d/a2
./a/b/c/c1
./a/b/c/c2
./a/b/c/a1
./a/b/c/a2
./a/b/b1
./a/b/b2
./a/b/a1
./a/b/a2
./a/a1
./a/a2
$ find a/b/c -path a/b/c -o -type d -prune -o -type f -a -name "a*" -print
a/b/c/a1
a/b/c/a2
```

Reply:
with `-prune` it's not readable anymore, imho. unless you consume some hallucinogens.

Reply:
Ha ha ha ha ha ha 
@telmich Your thoughts on this one?

Reply:
had one beer and i'm now trying to understand why `-prune`...

we use only one command and not two (second being `rm`)?

maybe we are splitting hairs here? :thinking:

Reply:
okay, i think i understand now, a bit. since we are only operating on one level, then we should use `-name`.

Reply:
replaced `-path` with `-name`

Reply:
`-prune` The primary shall always evaluate as true; it shall cause find not to descend the current pathname if it is a directory.

It can be used instead of `mindepth`, `maxdepth` which are not supported by POSIX.

But that is something @telmich  or @asteven should decide :)

Reply:
new morning and new ideas: `-name` should be replaced with `-regex`.

and i'll take look at `-prune`...

Reply:
@nico said " basically supporting bsds - realistically hpux, sunos, aix are not so much around anymore". so resolved. also replaced `-name` with `-regex`.